shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,6 +227,174 @@ async fn fetch_current_nodes(
     Ok(new_nodes)
 }
 
+enum ReplicationStrategy {
+    SimpleStrategy {
+        replication_factor: usize,
+    },
+    NetworkTopologyStrategy {
+        datacenter_replication_factor: HashMap<String, usize>,
+    },
+    Other,
+}
+
+mod system_keyspaces {
+    use super::*;
+    use serde_json::{Map, Value as JsonValue};
+    use std::str::FromStr;
+
+    pub async fn query(
+        connection: &CassandraConnection,
+        data_center: &str,
+    ) -> Result<HashMap<String, KeyspaceMetadata>> {
+        let (tx, rx) = oneshot::channel();
+
+        connection.send(Message::from_frame(
+            Frame::Cassandra(CassandraFrame{
+                version: Version::V4,
+                stream_id: 0,
+                tracing: Tracing::Request(false),
+                warnings: vec![],
+                operation: CassandraOperation::Query{
+                    query: Box::new(
+                        parse_statement_single(
+
+                            "SELECT keyspace_name, toJson(replication) AS replication FROM system_schema.keyspaces",
+                        )
+                    ),
+
+                    params: Box::new(QueryParams::default()),
+                }
+            })),
+            tx
+        )?;
+
+        let response = rx.await?.response?;
+        into_keyspaces(response, data_center)
+    }
+
+    fn into_keyspaces(
+        mut response: Message,
+        data_center: &str,
+    ) -> Result<HashMap<String, KeyspaceMetadata>> {
+        if let Some(Frame::Cassandra(frame)) = response.frame() {
+            match &mut frame.operation {
+                CassandraOperation::Result(CassandraResult::Rows { rows, .. }) => rows
+                    .iter_mut()
+                    .map(|row| build_keyspace(row, data_center))
+                    .try_collect(),
+                operation => Err(anyhow!(
+                    "keyspace query returned unexpected cassandra operation: {:?}",
+                    operation
+                )),
+            }
+        } else {
+            Err(anyhow!("Failed to parse keyspace query response"))
+        }
+    }
+
+    fn build_keyspace(
+        row: &mut Vec<MessageValue>,
```

Comment:
If we take ownership of row here it alleviates the worry of "oh but what if someone tries to use row after calling this build_keyspace."

![image](https://user-images.githubusercontent.com/5120858/199656369-d10cb1d9-832c-45f9-9fb8-fb22a564182e.png)